clang/lib/Frontend/CompilerInvocation.cpp

```diff
@@ -2091,6 +2091,12 @@ bool CompilerInvocation::ParseCodeGenArgs(CodeGenOptions &Opts, ArgList &Args,
   bool UsingProfile =
       UsingSampleProfile || !Opts.ProfileInstrumentUsePath.empty();
 
+  if (Args.hasArg(options::OPT_fdiagnostics_show_profile_count) &&
```

Comment:
If this option exists in `-cc1` only to emit a warning without being used for anything else, I suggest moving the diagnostic to the driver and removing the `-cc1` option.

Reply:
This option `options::OPT_fdiagnostics_show_profile_count` also appears in the `clang/lib/Driver/Toochains/clang.cpp` to enable the profile metadata appending to IR.

Reply:
Thanks for your comment! Remove `-cc1` option and migrate warning  generation to driver.